### PR TITLE
Refactor mount share storages recipes and create mount share storage resources

### PR DESCRIPTION
**Pull request by chenwany**
### Description of changes
Create mount share storage resources so they can be reused to support update share storages with mount and unmount actions

### Tests
* Manually test with cluster creation with EBS, EFS and FSX
* 
### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by demartinofra — COMMENTED** [review]
I like the usage of resources! well done.

A couple of minor high level optional comments:
1. I'd still keep one or more separate recipes to confine the storage logic so that we don't grow the base recipe. You could have something like fs_mount if you want to keep a single one.
2. from a logical point of view it probably makes more sense to have mount resources that handle single FS and iterate over the list of FS to mount outside of the resource. What I mean is having the loop in the recipe and then for each item add a new mount_* resource.

**Comment by chenwany** [comment]
> ouple of minor high level optional comments:
> 
> 1. I'd still keep one or more separate recipes to confine the storage logic so that we don't grow the base recipe. You could have something like fs_mount if you want to keep a single one.
> 2. from a logical point of view it probably makes more sense to have mount resources that handle single FS and iterate over the list of FS to mount outside of the resource. What I mean is having the loop in the recipe and then for each item add a new mount_* resource.

Thanks fdm. first comment is addressed.
For the second comment, yeah it also make sense to have mount resources that handle single FS. But for EBS, unmount action will require stop nfs before unmount all the volumes, and then start nfs after finish the process. If we want to make mount resources that handle single FS, then we need to take nfs stop and start steps outside mount resource.

**Review by demartinofra — DISMISSED** [review]
nice job 🚀

File: cookbooks/aws-parallelcluster-config/recipes/base.rb [hunk 1]
```diff
@@ -90,11 +90,8 @@
   mode '0644'
 end
 
-# Mount EFS directory with efs_mount recipe
-include_recipe 'aws-parallelcluster-config::efs_mount'
-
-# Mount FSx directory with fsx_mount recipe
-include_recipe 'aws-parallelcluster-config::fsx_mount'
+# Mount EFs, FSx
```
Comment (gmarciani) [inline cookbooks/aws-parallelcluster-config/recipes/base.rb]: Minor: EFs => EFS

File: cookbooks/aws-parallelcluster-config/recipes/fs_mount.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster
+# Recipe:: fs_mount
+#
+# Copyright:: 2013-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (gmarciani) [inline cookbooks/aws-parallelcluster-config/recipes/fs_mount.rb]: Minor: copyright year should be 2022

**Review by gmarciani — DISMISSED** [review]

File: cookbooks/aws-parallelcluster-config/recipes/fs_mount.rb (hunk 2 above)
Comment (chenwany) [reply cookbooks/aws-parallelcluster-config/recipes/fs_mount.rb]: Done

File: cookbooks/aws-parallelcluster-config/recipes/base.rb (hunk 1 above)
Comment (chenwany) [reply cookbooks/aws-parallelcluster-config/recipes/base.rb]: Done

**Review by gmarciani — APPROVED** [review]

**Pull request merged**